src/metaopt/io/database/mongodb.py

```diff
@@ -26,72 +26,77 @@ def __init__(self, dbhost='localhost', dbname='metaopt', dbport=27017, \
         self.dbname = str(dbname)
         self.username = str(username)
         self.password = str(password)
+        self.dbport = str(dbport)
 
-        # we might have a different port for the database port
-        # so here we do a check if the port is there
+        # try to initialize connection
+        # setting this as a function allows to make sure reconnection
+        # is properly done in the case of an unreliable network
+        self.uri = property(self.get_uri, self.set_uri, \
+                            None, 'This is the URI string of MongoDB')
+        self.conn = self.initiate_connection()
+        self.db = self.conn[self.dbname]
 
-        self.dbport = str(dbport)
+    def get_uri(self):
+        return self._uri
 
+    def set_uri(self, uri):
         # set the connection string
         # this should be mongodb://mongodb0.example.net:27017
         # according to MongoDB documentation
         # https://docs.mongodb.com/getting-started/python/client/
+        # simplest URI string is as followed
+        # self.uri = 'mongodb://' + self.dbhost + ':' + self.dbport
 
-        self.uri = 'mongodb://' + self.dbhost + ':' + self.dbport
-        #self.uri = "mongodb://%s:%s@%s:%s" % (quote_plus(self.username), \
-        #                                      quote_plus(self.password), \
-        #                                     self.dbhost, self.dbport)
-        # try to initialize connection
-        # setting this as a function allows to make sure reconnection
-        # is properly done in the case of an unreliable network
-
-        self.conn = self.initiate_connection()
-        self.db = self.conn[self.dbname]
+        uri = "mongodb://%s:%s@%s:%s" % (quote_plus(self._username), \
+                                         quote_plus(self._password), \
+                                         self._dbhost, self._dbport)
+        self._uri = uri
```

Comment:
`uri` passed to the setter is not used. Do we support editing dbhost, dbport, etc, or editing uri? Or both? I think we should support both, but for simplicity I would only support setting dbhost, dbport and others for now. If we allow setting the uri, than we need to parse it and set dbhost and others accordingly. We can keep that for later.